frontend/src/components/UserList/UserList.tsx

```diff
@@ -1,60 +1,56 @@
-import { DataGrid } from '@mui/x-data-grid';
-import React from 'react';
-
-import { UserListColumns } from '.';
-import { randomEmail, randomName } from '@mui/x-data-grid-generator';
-
-const SampleRows = [
-  {
-    id: 1,
-    avatar: null,
-    name: randomEmail(),
-    email: randomEmail(),
-  },
-  {
-    id: 2,
-    avatar: 'https://picsum.photos/200',
-    name: randomEmail(),
-    email: randomEmail(),
-  },
-  {
-    id: 3,
-    avatar: 'https://picsum.photos/200',
-    name: randomEmail(),
-    email: randomEmail(),
-  },
-  {
-    id: 4,
-    avatar: 'https://picsum.photos/200',
-    name: randomEmail(),
-    email: randomEmail(),
-  },
-  {
-    id: 5,
-    avatar: 'https://picsum.photos/200',
-    name: randomEmail(),
-    email: randomEmail(),
-  },
-];
-
-interface Props {}
-
-export const UserList = (props: Props) => {
+import { useEffect } from 'react';
+
+import { UserListData } from '.';
+
+import { fetchUsers, followUser } from '../../actions';
+import { StoreState } from '../../reducers';
+import { connect } from 'react-redux';
+import { User } from '../AdminUser';
+import { GridRenderCellParams } from '@mui/x-data-grid';
+
+interface Props {
+  fetchUsers: Function;
+  followUser: Function;
+  usersWithFollows: User[];
+}
+
+export const UserList = ({
+  fetchUsers,
+  followUser,
+  usersWithFollows,
+}: Props): JSX.Element => {
+  useEffect(() => {
+    fetchUsers();
+    return () => {};
+  }, []);
+
+  const handleFollowClick = (e: any, params: GridRenderCellParams) => {
+    let rowId = params.id;
+    followUser(rowId);
+  };
+
   return (
     <div className="container mx-auto mt-20 h-screen">
       <div className="flex mx-auto h-4/6 w-1/2 ">
         <div className="flex-grow">
-          <DataGrid
-            rows={SampleRows}
-            columns={UserListColumns}
-            pagination
-            pageSize={10}
-            rowsPerPageOptions={[10, 25, 50]}
+          <UserListData
+            handleFollowClick={handleFollowClick}
+            usersWithFollows={usersWithFollows}
           />
         </div>
       </div>
     </div>
   );
 };
 
-export default UserList;
+const mapStateToProps = ({
+  usersData,
+}: StoreState): { usersWithFollows: User[] } => {
+  const users = usersData.data || [];
+  const usersWithFollows = users.map((user) => {
+    return { ...user };
+  });
```

Comment:
Did you know that you can do this?
```suggestion
  const usersWithFollows = users.map((user) => ({ ...user }));
```

Reply:
yes yes, i have it that way in preparation for supplying additional data to the user object

Reply:
latest state of the codeblock

![image](https://user-images.githubusercontent.com/95029803/148726216-6cf2551f-52ca-41eb-adae-a55321e27f28.png)